Source/WebKit/Platform/IPC/HandleMessage.h

```diff
@@ -123,118 +123,80 @@ void logReply(const Connection& connection, MessageName messageName, const T&...
 #endif
 }
 
-// Dispatch functions with no reply arguments.
-
-template <typename C, typename MF, typename ArgsTuple, size_t... ArgsIndex>
-void callMemberFunctionImpl(C* object, MF function, ArgsTuple&& args, std::index_sequence<ArgsIndex...>)
+template<typename TupleType, typename F>
+void applyTuple(TupleType&& tuple, const F& functor)
 {
-    (object->*function)(std::get<ArgsIndex>(std::forward<ArgsTuple>(args))...);
+    [&]<size_t... Indices>(std::index_sequence<Indices...>)
+    {
+        functor(std::get<Indices>(std::forward<TupleType>(tuple))...);
+    }(std::make_index_sequence<std::tuple_size_v<TupleType>> { });
 }
 
-template<typename C, typename MF, typename ArgsTuple, typename ArgsIndices = std::make_index_sequence<std::tuple_size<ArgsTuple>::value>>
-void callMemberFunction(ArgsTuple&& args, C* object, MF function)
+// Dispatch functions with no reply arguments.
+
+template<typename C, typename MF, typename ArgsTuple>
```

Comment:
FWIW I think stack overflow gives:
```
template<typename F, typename T, typename U>
decltype(auto) apply_invoke(F&& func, T&& first, U&& tuple) {
    return std::apply(std::forward<F>(func), std::tuple_cat(std::forward_as_tuple(std::forward<T>(first)), std::forward<U>(tuple)));
}```

Comparing to that, at least your MF could be an universal reference?

Reply:
`std::apply` has to cover any callable object, including lambdas. MF I think will always be a pointer to a class method, perhaps I can type it like that. But a universal reference wouldn't do harm.

Reply:
I ended up using `T U::*`, they're member function pointers, so the relevant class and the function type can be immediately deduced this way.